core/src/mindustry/world/blocks/storage/CoreBlock.java

```diff
@@ -164,6 +168,18 @@ public void onProximityUpdate(){
             }
         }
 
+        @Override
+        public void draw(){
+            if(team == Team.sharded){
+                Draw.rect(block.region, x, y, block.rotate ? rotdeg() : 0);
+            }else{
+                Draw.rect(baseRegion, x, y, block.rotate ? rotdeg() : 0);
+                Draw.color(team.color);
+                Draw.rect(topRegion, x, y, block.rotate ? rotdeg() : 0);
+                Draw.color();
+            }
+        }
+
```

Comment:
not sure if you want to have `, block.rotate ? rotdeg() : 0` there or not ^ (modding support?)